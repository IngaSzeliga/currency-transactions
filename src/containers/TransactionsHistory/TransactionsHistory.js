import React from "react";
import PropTypes from "prop-types";
import Transaction from "../../components/Transaction";
import { countryRegion, PLN_NAME } from "../../config";
import "./TransactionsHistory.scss";

const TransactionsHistory = props => {
  const { transactions, deleteTransaction } = props;
  let totalTransaction = 0;
  for (var i = 0; i < transactions.length; i++) {
    totalTransaction = totalTransaction + transactions[i].convertedAmount;
  }
  return (
    <div className="transactions-history-container">
      <h2 className="transactions-history-container-title">
        Transaction History:
      </h2>
      {transactions.length === 0
        ? "No transaction yet"
        : `Sum of transactions: ${totalTransaction.toLocaleString(
            countryRegion,
            {
              style: "currency",
              currency: PLN_NAME
            }
          )}`}
      {transactions.map(transaction => {
        const { id, title, amount, convertedAmount } = transaction;
        return (
          <Transaction
            key={id}
            id={id}
            title={title}
            amount={amount}
            convertedAmount={convertedAmount}
            deleteTransaction={deleteTransaction}
          />
        );
      })}
    </div>
  );
};

TransactionsHistory.propTypes = {
  transactions: PropTypes.array.isRequired,
  deleteTransaction: PropTypes.func.isRequired
};

export default React.memo(TransactionsHistory);
