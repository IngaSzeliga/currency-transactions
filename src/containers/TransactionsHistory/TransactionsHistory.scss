@import "../../config/sizes.scss";

.transactions-history-container {
  padding: 40px;
  width: 90%;
  @media only screen and (min-width: $desktop) {
    width: 40%;
  }
}
