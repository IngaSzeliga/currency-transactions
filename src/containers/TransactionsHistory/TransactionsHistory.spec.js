import React from "React";
import { shallow } from "enzyme";
import TransactionsHistory from ".";

describe("TransactionsHistory component", () => {
  const component = shallow(
    <TransactionsHistory
      transactions={[
        {
          id: "2caf5711-2913-4e41-95db-088625acf1bf",
          title: "Name",
          amount: 13,
          convertedAmount: "54,99 zł"
        }
      ]}
      deleteTransaction={jest.fn()}
    />
  );

  it("has working component", () => {
    expect(component).toBeDefined();
  });

  it("expects component to never be changed", () => {
    expect(component).toMatchSnapshot();
  });

  it("expects component to never be changed with no transactions", () => {
    const component = shallow(
      <TransactionsHistory transactions={[]} deleteTransaction={jest.fn()} />
    );
    expect(component).toMatchSnapshot();
  });
});
