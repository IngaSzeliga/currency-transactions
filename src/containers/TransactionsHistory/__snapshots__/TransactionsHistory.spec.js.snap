// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionsHistory component expects component to never be changed 1`] = `
<div
  className="transactions-history-container"
>
  <h2
    className="transactions-history-container-title"
  >
    Transaction History:
  </h2>
  Sum of transactions: 054,99 zł
  <Memo(Transaction)
    amount={13}
    convertedAmount="54,99 zł"
    deleteTransaction={[MockFunction]}
    id="2caf5711-2913-4e41-95db-088625acf1bf"
    key="2caf5711-2913-4e41-95db-088625acf1bf"
    title="Name"
  />
</div>
`;

exports[`TransactionsHistory component expects component to never be changed with no transactions 1`] = `
<div
  className="transactions-history-container"
>
  <h2
    className="transactions-history-container-title"
  >
    Transaction History:
  </h2>
  No transaction yet
</div>
`;
